src/app/DIAmeterApplication.h

```diff
@@ -57,31 +69,82 @@ class DIAmeterApplication : public CruxApplication {
 	const ProteinVec& proteins, ///< proteins corresponding with peptides
 	const vector<const pb::AuxLocation*>& locations,  ///< auxiliary locations
 	TideMatchSet* matches, ///< object to manage PSMs
-	map<int, pair<double*, double*>>* ms1scan_intensity_rank_map,
+	ObservedPeakSet* observed,
+	// map<int, pair<double*, double*>>* ms1scan_intensity_rank_map,
+	map<int, boost::tuple<double*, double*, double*, int>>* ms1scan_mz_intensity_rank_map,
+	map<int, boost::tuple<double, double>>* ms1scan_slope_intercept_map,
+	// map<int, double*>* ms2scan_intensity_map,
+	map<int, boost::tuple<double*, double*, int>>* ms2scan_mz_intensity_map,
 	map<string, double>* peptide_predrt_map
   );
 
-  void computePrecIntRank(
+  void computePrecIntRankOld(
 	const vector<TideMatchSet::Arr::iterator>& vec,
 	const ActivePeptideQueue* peptides,
 	const double* intensity_rank_arr,
 	map<TideMatchSet::Arr::iterator, boost::tuple<double, double, double>>* intensity_map,
 	int charge
   );
+  void computePrecIntRankNew(
+  	const vector<TideMatchSet::Arr::iterator>& vec,
+  	const ActivePeptideQueue* peptides,
+  	const double* mz_arr,
+	const double* intensity_arr,
+  	const double* intensity_rank_arr,
+	boost::tuple<double, double> slope_intercept_tp,
+	int peak_num,
+  	map<TideMatchSet::Arr::iterator, boost::tuple<double, double, double>>* intensity_map,
+  	map<TideMatchSet::Arr::iterator, boost::tuple<double, double, double>>* logrank_map,
+  	int charge
+  );
+
+  void computePrecFragCoeluteOld(
```

Comment:
same question regarding "old" and "new". Either merge into one fxn (with flag) or comment saying why we need 2.